katharsis-core/src/main/java/io/katharsis/utils/PropertyUtils.java

```diff
@@ -310,12 +310,9 @@ private void setPropertyValue(Object bean, String fieldName, Object value)
     @SuppressWarnings("unchecked")
     private Object prepareValue(Object value, Class<?> fieldClass) {
```

Comment:
Unless there is a reason, it would be best to return a list in either case. This will naturally preserve order. 

But, if your entity has a member that's a set, keep in mind that json doesn't have sets, it has only arrays. Unordered and indexable. Sets are inherently non indexable. That means PATCH will always be an issue under this condition.  

I understand the need for order what I don't get is why this code converts set to list and list to set? 

But, you didn't write the original code. I can approve the change because it's safe enough.